arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -454,14 +475,18 @@ public inline fun <A, B> Iterable<A>.traverse(f: (A) -> Option<B>): Option<List<
 }
 
 @Deprecated(
-  "sequenceOption is being renamed to sequence to simplify the Arrow API",
-  ReplaceWith("sequence()", "arrow.core.sequence")
+  "The sequence extension function is being deprecated in favor of the option DSL.",
+  ReplaceWith("let { l -> option<List<A>> { l.bindAll() } }", "arrow.core.raise.option")
 )
 public fun <A> Iterable<Option<A>>.sequenceOption(): Option<List<A>> =
-  sequence()
+  let { l -> option { l.bindAll() } }
 
+@Deprecated(
+  "The sequence extension function is being deprecated in favor of the option DSL.",
+  ReplaceWith("let { l -> option<List<A>> { l.bindAll() } }", "arrow.core.raise.option")
+)
 public fun <A> Iterable<Option<A>>.sequence(): Option<List<A>> =
-  traverse(::identity)
+  let { l -> option { l.bindAll() } }
```

Comment:
the let is unnecessary here, right?

Reply:
It's so that the `ReplaceWith` is inline with the implementation body. `let` should incur any perf penalty due to `@InlineOnly`.